### PR TITLE
fix: return an error when extracting record name

### PR DESCRIPTION
**Pull request by ldez**
return an error when extracting record name instead of panic

### CODE REVIEW

File: providers/dns/constellix/constellix.go [hunk 1]
```diff
@@ -263,6 +269,12 @@ func containsValue(record *internal.Record, value string) bool {
...
-func getRecordName(fqdn, authZone string) string {
-	return fqdn[0 : len(fqdn)-len(authZone)-1]
+func getRecordName(fqdn, authZone string) (string, error) {
+	end := len(fqdn) - len(authZone) - 1
+
+	if len(fqdn) < end || end < 0 {
+		return "", fmt.Errorf("%d is lower than the length of the fqdn (fqdn: %s, authZone: %s)", end, fqdn, authZone)
+	}
+
+	return fqdn[0:end], nil
```
Comment (dmke) [inline providers/dns/constellix/constellix.go:279]: This function is repeated four times. Can we move it into an utilities package? `dns/providers/internal` already exists, so having a "records" sub package there wouldn't hurt that much.

**Review by dmke — CHANGES_REQUESTED** [review]

File: providers/dns/constellix/constellix.go (hunk 1 above)
Comment (ldez) [reply providers/dns/constellix/constellix.go:279]: The name is generic, we have several different implementations, so I don't want to create confusion.

Comment (dmke) [reply providers/dns/constellix/constellix.go:279]: I didn't know we had a "generic" package. For me, the important bit is that there are four copy-pasted implementations of non-trivial behaviour without tests.

Comment (dmke) [reply providers/dns/constellix/constellix.go:279]: Q: Is it possible to have `fqdn == "foo.example.com."` and `authZone == "example.org"`? 

If so, this function returns `"foo", nil`, and I'm not sure if this is another bug lingering here.

Comment (dmke) [reply providers/dns/constellix/constellix.go:279]: Ah, I believe I can answer that myself: `authZone` is the domain containing the SOA record for `fqdn`. IIRC, this requires `fqdn` to be a sub-domain of `authZone`.

Comment (ldez) [reply providers/dns/constellix/constellix.go:279]: I'm not a huge fan of "utils" package. I have to think.

Comment (ldez) [reply providers/dns/constellix/constellix.go]: I created a function `ExtractSubDomain` inside the `dns01` package.

**Review by dmke — APPROVED** [review]
LGTM

**Pull request merged**